pkg/apis/kubebind/v1alpha1/permissionclaims_types.go

```diff
@@ -0,0 +1,273 @@
+// consumer to the provider cluster or vice versa. Exceptions can be specified
+// for individual JSON Paths to be owned by the other side. Metadata in general
+// is not synced. Exceptions for labels and annotations can be specified.
+//
+// Ownership of an object is determined by the `kube-bind.io/owner` annotation.
+// Objects on the consumer cluster are owned by the consumer by default. Objects
+// on the provider cluster are owned by the provider by default. The annotation
+// is only set if the object is owned by the respective other side.
+//
+// +kubebuilder:validation:XValidation:rule="!(has(self.autoDonate) && self.autoDonate && has(self.autoAdopt) && self.autoAdopt)",message="donate and adopt are mutually exclusive"
+type PermissionClaim struct {
+	GroupResource `json:","`
+
+	// version is the version of the claimed resource.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength:=1
+	Version string `json:"version"`
+
+	// selector restricts which objects of the given resource are being claimed.
+	// If unset, all objects across all namespaces are being claimed, both
+	// consumer- and provider owned.
+	//
+	// +optional
+	// +kubebuilder:default:={}
+	Selector *ResourceSelector `json:"selector,omitempty"`
+
+	// read claims read access for the provider to matching objects, excluding
+	// labels and annotations by default. Read access is realized by syncing
+	// the objects or fields from the consumer cluster to the provider cluster.
+	//
+	// +optional
+	// +kubebuilder:default={}
+	Read *PermissionClaimReadOptions `json:"read,omitempty"`
+
+	// create determines whether the provider can create new objects in the
+	// consumer cluster by syncing a provider-owned source to the consumer
+	// cluster. Created objects on the consumer cluster are marked as owned by
+	// the provider by default.
+	//
+	// Note that create permissions do not imply update permissions.
+	//
+	// +optional
+	Create *PermissionClaimCreateOptions `json:"create,omitempty"`
+
+	// autoAdopt set to true means that objects created by the consumer on the
+	// consumer cluster are immediately adopted by the provider by syncing them
+	// to the provider side and then marking them as owned by the provider on
+	// both sides.
+	//
+	// autoAdopt is mutually exclusive with autoDonate.
+	//
+	// +optional
+	AutoAdopt bool `json:"autoAdopt,omitempty"`
+
+	// autoDonate set to true means that a newly created object synced from the
+	// provider to the consumer cluster is immediately donated to the consumer
+	// by marking it as owned by the consumer on both sides.
+	//
+	// autoDonate is mutually exclusive with autoAdopt.
+	//
+	// +optional
+	AutoDonate bool `json:"autoDonate,omitempty"`
+
+	// onConflict determines how conflicts between objects on the consumer
+	// and provider clusters will be resolved.
+	//
+	// +optional
+	// +kubebuilder:default:={}
+	OnConflict *PermissionClaimOnConflictOptions `json:"onConflict,omitempty"`
```

Comment:
Somehow this does not look right. You will have conflict on Create/Update only. Now this still being optional but implied to get and Deletes too. Some logical shuffling might be needed. Like Clients logic: `Delete(ctx context.Context, name string, opts v1.DeleteOptions) error` where DeleteOptions are only for delete struct and we can reuse them inside.